FEA add ValidationCurveDisplay in model_selection module (#25120)
scikit-learn/scikit-learn: doc/modules/learning_curve.rst

  >>> X, y = X[indices], y[indices]

  >>> train_scores, valid_scores = validation_curve(
  ...     SVC(kernel="linear"), X, y, param_name="C", param_range=np.logspace(-7, 3, 3),
  ... )
  >>> train_scores
  array([[0.90..., 0.94..., 0.91..., 0.89..., 0.92...],
         [0.9... , 0.92..., 0.93..., 0.92..., 0.93...],
         [0.97..., 1...   , 0.98..., 0.97..., 0.99...]])
  >>> valid_scores
  array([[0.9..., 0.9... , 0.9... , 0.96..., 0.9... ],
         [0.9..., 0.83..., 0.96..., 0.96..., 0.93...],
         [1.... , 0.93..., 1....  , 1....  , 0.9... ]])

.. figure:: ../auto_examples/model_selection/images/sphx_glr_plot_validation_curve_001.png
   :target: ../auto_examples/model_selection/plot_validation_curve.html
   :align: center

>    :scale: 50%

Reviewer: This should also be moved at the end of the session to match the "in the plot below" reference, otherwise this is very confusing.

> .. figure:: ../auto_examples/model_selection/images/sphx_glr_plot_validation_curve_001.png
>    :target: ../auto_examples/model_selection/plot_validation_curve.html
>    :align: center
>    :scale: 50%

Reviewer: ```suggestion
```

   :align: center
   :scale: 50%

> If you intend to plot the validation curves only, the class

Reviewer: I would move this figure right after the code snippet above and move the `If the training score and the validation score are both low` after the new plot.

Otherwise, it's weird: we compute the validation curves of the `C` parameter of a `SVC(kernel="linear")` model on iris using the `validation_curve` function, then discuss how to analyze the values using the figure of a plot that studies the `gamma` parameter of `SVC(kernel="rbf")` on another dataset from an example and then come back with another plot on the original model and dataset which is quite confusing.

It's based to avoid the back and forth switching between the 2 model/dataset pairs by reordering those paragraphs as suggested.

> :class:`~sklearn.model_selection.ValidationCurveDisplay` will be easier to use.

Reviewer: Let's not use "will" when there is no future.

```suggestion
:class:`~sklearn.model_selection.ValidationCurveDisplay` is more direct than
using matplotlib manually on the results of a call to :func:`validation_curve`.
```

You can use the method
:meth:`~sklearn.model_selection.ValidationCurveDisplay.from_estimator` similarly
to :func:`validation_curve` to generate and plot the validation curve:

.. plot::
   :context: close-figs
   :align: center

      from sklearn.datasets import load_iris
      from sklearn.model_selection import ValidationCurveDisplay
      from sklearn.svm import SVC
      from sklearn.utils import shuffle
      X, y = load_iris(return_X_y=True)
      X, y = shuffle(X, y, random_state=0)
      ValidationCurveDisplay.from_estimator(

>          SVC(kernel="linear"), X, y, param_name="C", param_range=np.logspace(-7, 3, 10))

Reviewer: Any color you like... as long as it's black.

```suggestion
         SVC(kernel="linear"), X, y, param_name="C", param_range=np.logspace(-7, 3, 10)
     )
```

overfitting, and a working model are shown in the in the plot below where we vary

> the parameter :math:`\gamma` of an SVM on the digits dataset.

Reviewer: ```suggestion
the parameter `gamma` of an SVM on the digits dataset.
```
